### ISSUE TITLE
comments rendering : case of children of not approved comment

### ISSUE THREAD

**Issue by er314** [ryancramerdesign/ProcessWire#1334]
Approved comments are not rendered (not searched by getReplies) if they are children of a comment marked as spam.
Is this intended ?
I agree that such a scenario is not the most thrilling ever, but anyway, I guess we have 3 options here : 
1. as currently : rendering ignores any comments that are children of a spam or a pending comment
-> rendering is then not consistent with comments status ; consistency issue too for approved comments counting
2. when a comment is marked as spam (or pending), then mark also its children the same
-> this is somewhat violent ; but at least it's consistent with the rendering
3. render all approved comments ; if a comment is child of a spam or pending comment, then render it at level 0 + add a specific class & text to it (thus allowing developer to perform dedicated styling on it)
-> my preference goes to 3., but maybe I'm overlooking a better option

**Comment by ryancramerdesign** [comment]
Yes this is intended. I understand what you are saying. My opinion here is that we shouldn't be rendering comments that have an unapproved/spam parent because that comment then lacks proper context. It's a reply to something where the source isn't visible. I think it's also pretty unusual to have an unapproved comment with approved children, because the unapproved comment would have never been displayed for anyone to reply to. It would basically have to be a comment that was approved and then later got unapproved. If the admin thinks replies to an unapproved comment are important enough to warrant their visibility, they can always move the comment out of the unapproved parent. 

The comments output logic is something within the control of the developer. Our default rendering uses the intended logic, but comments are a WireArray like any other, and someone can output it however they prefer. Originally when building this module, I thought that most would do this, like they might for a PageArray. Though admittedly I haven't seen many that do this to date. Maybe longer term more hooks to the built-in rendering behaviors will be a welcome addition.

**Comment by er314** [comment]
Thanks for (yet another :-) detailed reply.
Yes, comments rendering is within the control of the developer, I actually did code my own rendering, but up to now I keep relying on the core getReplies() function, and in regards to this point, instead of coding myself a custom version, I first felt that its "fix" would be beneficial to the community, hence this open issue.
I agree with your arguments, except "the unapproved comment would have never been displayed for anyone to reply to" : if the comments field is set in no-moderation mode, actually this case will very likely occur, where marking a comment as spam will be performed after the fact.

FYI, I have noticed that disqus behaviour is the following, which is the approach I think I prefer, out of respect for the authors of children comments (even though, as you say, you then lack proper context), see screenshot.

cheers

![image](https://cloud.githubusercontent.com/assets/8220747/9319990/0ef1b1e0-4555-11e5-9072-81e3964178fc.png)
